src/main/kotlin/com/terraformation/backend/accelerator/db/ApplicationStore.kt

```diff
@@ -0,0 +1,266 @@
+package com.terraformation.backend.accelerator.db
+
+import com.terraformation.backend.accelerator.model.ExistingApplicationModel
+import com.terraformation.backend.auth.currentUser
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.db.OrganizationNotFoundException
+import com.terraformation.backend.db.accelerator.ApplicationId
+import com.terraformation.backend.db.accelerator.ApplicationStatus
+import com.terraformation.backend.db.accelerator.tables.references.APPLICATIONS
+import com.terraformation.backend.db.accelerator.tables.references.APPLICATION_HISTORIES
+import com.terraformation.backend.db.default_schema.OrganizationId
+import com.terraformation.backend.db.default_schema.ProjectId
+import com.terraformation.backend.db.default_schema.tables.daos.CountriesDao
+import com.terraformation.backend.db.default_schema.tables.daos.OrganizationsDao
+import com.terraformation.backend.gis.CountryDetector
+import com.terraformation.backend.log.perClassLogger
+import jakarta.inject.Named
+import java.time.InstantSource
+import org.jooq.Condition
+import org.jooq.DSLContext
+import org.jooq.impl.DSL
+import org.locationtech.jts.geom.Geometry
+
+@Named
+class ApplicationStore(
+    private val clock: InstantSource,
+    private val countriesDao: CountriesDao,
+    private val countryDetector: CountryDetector,
+    private val dslContext: DSLContext,
+    private val organizationsDao: OrganizationsDao,
+) {
+  private val log = perClassLogger()
+
+  fun fetchOneById(applicationId: ApplicationId): ExistingApplicationModel {
+    requirePermissions { readApplication(applicationId) }
+
+    return fetchByCondition(APPLICATIONS.ID.eq(applicationId)).firstOrNull()
+        ?: throw ApplicationNotFoundException(applicationId)
+  }
+
+  fun fetchByProjectId(projectId: ProjectId): List<ExistingApplicationModel> {
+    requirePermissions { readProject(projectId) }
```

Comment:
This allows org members below admin to readApplication, which is more relaxed than `readApplication` I think

Reply:
Good catch. Added code to filter the results.